module-sys/SystemManager/SystemManagerCommon.cpp

```diff
@@ -597,9 +578,13 @@ namespace sys
             return MessageNone{};
         });
 
+        /* Workaround for hardware issues with power supply. With USB cable connected the phone cannot
+         * restart, so detect such state and perform reboot when right function button is pressed.
+         * This way for the user the behavior looks as if the phone was normally turned on from shutdown
+         * state. */
         connect(sevm::KbdMessage(), [&](Message *) {
-            // we are in shutdown mode - we received that there was red key pressed -> we need to reboot
-            if (state == State::Shutdown) {
+            if ((state == State::Shutdown) && (Store::Battery::get().state != Store::Battery::State::Discharging)) {
```

Comment:
Is it possible to be in such a state where USB cable is plugged and system is reporting 'Discharging' status? The obvious one that comes to mind is some kind of error condition but is there anything else?

Reply:
Apart from that, `Store::Battery::get()` is something I would like to discourage from using it. At the same time, I'm aware this concept is being used throughout the system(sadly).

Reply:
My bad, it was a change I left there by mistake while working on some other issue, removed it now ;)